lua/spectre_common.lua

```diff
@@ -291,6 +293,27 @@ local function get_target_uri(request_uri, request_headers)
 
 end
 
+local function configure_custom_hooks(custom_module)
+    ngx.ctx.custom_hooks = {}
+    -- If a custom module argument is provided, try and use it, otherwise use defaults
+    if custom_module ~= nil then
+        local custom_mod = custom_hooks[custom_module]
+        if custom_mod ~= nil then
+            -- overwrite the default hooks with the custom hooks from the module
+
+            -- if custom hook - custom_cache_key_fn - defined
+            if custom_mod.custom_cache_key_fn ~= nil then
+                ngx.ctx.custom_hooks.cache_key = custom_mod.custom_cache_key_fn
+            end
+        else
+            ngx.ctx.custom_hooks.cache_key = custom_hooks.defaults.custom_cache_key_fn
+        end
```

Comment:
if custom_module does not has  `custom_cache_key_fn` but something else... we will get an exception becuase `ngx.ctx.custom_hooks.cache_key` would be nil.